src/main/java/ch/njol/skript/expressions/ExprTypeOf.java

```diff
@@ -61,14 +54,17 @@ public Class<?> getReturnType() {
 		return EntityData.class.isAssignableFrom(returnType) ? EntityData.class
 			: ItemType.class.isAssignableFrom(returnType) ? ItemType.class
 			: PotionEffectType.class.isAssignableFrom(returnType) ? PotionEffectType.class
-			: BlockData.class.isAssignableFrom(returnType) ? ItemType.class : Object.class;
+			: BlockData.class.isAssignableFrom(returnType) ? ItemType.class
+			: EnchantmentType.class.isAssignableFrom(returnType) ? Enchantment.class
+			: Object.class;
 	}
```

Comment:
Is this still safe? Won't the expression type just be the converted `AnyTyped` for BlockData and potions?

Reply:
yeah this draft basically needs to be redone, I think I'll be closing it for now